src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -219,13 +219,32 @@ public PackageBuilder GetPackageBuilder(IPackTaskRequest<IMSBuildItem> request)
                     .ToDictionary(msbuildItem => msbuildItem.Identity,
                     msbuildItem => msbuildItem.GetProperty("ProjectVersion"), PathUtility.GetStringComparerBasedOnOS());
             }
+
+            var aliases = new Dictionary<string, string>();
+            foreach (var tfm in assetsFile.PackageSpec.TargetFrameworks)
+            {
+                aliases[tfm.TargetAlias] = tfm.FrameworkName.GetShortFolderName();
+            }
+
             var nuGetFrameworkComparer = new NuGetFrameworkFullComparer();
             var frameworksWithSuppressedDependencies = new HashSet<NuGetFramework>(nuGetFrameworkComparer);
             if (request.FrameworksWithSuppressedDependencies != null && request.FrameworksWithSuppressedDependencies.Any())
             {
                 frameworksWithSuppressedDependencies =
                     new HashSet<NuGetFramework>(request.FrameworksWithSuppressedDependencies
-                    .Select(t => NuGetFramework.Parse(t.Identity)).ToList(), nuGetFrameworkComparer);
+                    .Select(t =>
+                    {
+                        string translated = null;
+                        var succeeded = aliases.TryGetValue(t.Identity, out translated);
+                        if (succeeded)
```

Comment:
style: 

You can save a few lines and imo improve readability by writing something like: 
```cs
                        if (aliases.TryGetValue(t.Identity, out string translated))
                        {
                            return NuGetFramework.Parse(translated);
                        }
```